coalib/settings/FunctionMetadata.py

```diff
@@ -160,3 +160,49 @@ def from_function(cls, func, omit=frozenset()):
                    non_optional_params=non_optional_params,
                    optional_params=optional_params,
                    omit=omit)
+
+    @classmethod
+    def merge(cls, *metadatas):
+        """
+        Merges signatures of ``FunctionMetadata`` objects.
+
+        Parameter (either optional or non-optional) and non-parameter
+        descriptions are merged from left to right, meaning the right hand
+        metadata overrides the left hand one.
+
+        :param metadatas:
+            The sequence of metadatas to merge.
+        :raises ValueError:
+            Raised when less than 2 metadatas are provided.
+        :return:
+            A ``FunctionMetadata`` object containing the merged signature of
+            all given metadatas.
+        """
+        # Collect the metadatas, as we operate on them more often and we want
+        # to support arbitrary sequences.
+        metadatas = tuple(metadatas)
+        if len(metadatas) < 2:
+            raise ValueError("Less than 2 metadata objects provided.")
```

Comment:
this should work very well with one metadata object shouldn't it? Or zero. Less special cases for outsiders to handle.